Document how to create single-user dandihub images (#188)
dandi/dandi-docs: docs/50_hub.md

If you need additional software installed in the image, you can add a server image that will be made available for all users in the `Server Options` menu.  Add a server image by updating the `profileList` in the [JupyterHub config file](https://github.com/dandi/dandi-hub/blob/do-eks/helm/jupyterhub/dandihub.yaml) and submitting a pull request to the [dandi-hub repository](https://github.com/dandi/dandi-hub).  Once the pull request is merged, the DANDI team will redeploy JupyterHub and the image will be available.

> See https://github.com/dandi/dandi-hub/pull/233 as an example of the changes you need.
> 

Reviewer: Let's remove this link since there is a bug in this code.

```suggestion
```

> The following command will build and run a new image, which can be viewed locally in the browser at 127.0.0.1:8888/
> 

Reviewer: ```suggestion
1. Fork and clone the [dandi-hub](https://github.com/dandi/dandi-hub) repository.
2. Add a Dockerfile to the [images](https://github.com/dandi/dandi-hub/tree/main/images) directory.
3. Test the Dockerfile with the following commands to build and run the new image, which can be viewed locally in the browser at 127.0.0.1:8888/
```

```sh

> docker build -f $(CONTAINERFILE) -t dandihub-dev:latest .

Reviewer: ```suggestion
docker build -f <Your-Dockerfile> -t dandihub-dev:latest .
```

Reviewer 2: "lazy me" likes when I can just cut/paste commands.  Here if `CONTAINERFILE` variable was defined above, it would satisfy my desires -- it would also make it "easier" to customize since it is just that -- remove/place value.

Then here it should be

```suggestion
docker build -f "$(CONTAINERFILE)" -t dandihub-dev:latest .
```

to be kosher and no navigating relatively long line is needed to edit it etc.

Reviewer 2: since to be used only once, defining a variable might be not really warranted though

Reviewer: `Dockerfile.[gpu].<project-name>` - see other [Dockerfiles](https://github.com/dandi/dandi-hub/tree/main/images).

docker run --rm -p 8888:8888 --name dev_jupyterlab dandihub-dev:latest start-notebook.sh --NotebookApp.token=""

> ```

Reviewer: May need to indent this section.

> 
> To add the new image to the test CI, and to the Dandi Archive Dockerhub, add it to the `include`
> matrix of both `.github/workflows/docker-push.yaml` and `.github/workflows/docker-test.yaml`.
> 
> You can add the image to server options by adding options to `envs/shared/jupyterhub.yaml`
> 

Reviewer: ```suggestion
4. Add the Dockerfile to the `include` matrix of both the [docker-push.yaml](https://github.com/dandi/dandi-hub/blob/main/.github/workflows/docker-push.yaml) and [docker-test.yaml](https://github.com/dandi/dandi-hub/blob/main/.github/workflows/docker-test.yaml) files.  This will allow the image to be built when new pull requests are opened and pushed to the [DANDI Archive Docker Hub](https://hub.docker.com/u/dandiarchive) when the pull requests are merged. 
5. Add the image to the server options by updating the [jupyterhub.yaml](https://github.com/dandi/dandi-hub/blob/main/envs/shared/jupyterhub.yaml) file.
```
